modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/podcast/UserEpisodeManager.kt

```diff
@@ -398,9 +399,11 @@ class UserEpisodeManagerImpl @Inject constructor(
             val episode = findEpisodeByUuid(it)
             if (episode != null) {
                 if (!episode.isDownloaded) {
-                    // File deleted from server
-                    playbackManager.removeEpisode(episode, source = SourceView.UNKNOWN, userInitiated = false)
-                    userEpisodeDao.delete(episode)
+                    // Remove file not found on server only if it is not added to up-next to prevent it from being removed from up-next (Github Issue#356)
+                    if (!upNextDao.containsEpisode(episode.uuid)) {
+                        playbackManager.removeEpisode(episode, source = SourceView.UNKNOWN, userInitiated = false)
+                        userEpisodeDao.delete(episode)
+                    }
```

Comment:
If a user episode is not found on the server, it is now deleted only if it is not present in the up-next queue.

Reply:
Hopefully that doesn't mean once episodes are removed from upnext the non uploaded file will get deleted, I intentionally have it set so none of my files gets deleted unless I intend to delete them 😅